src/pydoc_markdown/contrib/renderers/mkdocs.py

```diff
@@ -110,6 +111,27 @@ class MkdocsRenderer(Renderer, Server, Builder):
   #: ```
   server_port: t.Optional[int] = None
 
+  #: Find the reference in the current module.
+  #: Works like the default resolver but with a little bit of enhancement.
+  #: Default to `True`.
+  crossref_local: bool = True
+  
+  #: Find the reference in all of the imported modules.
+  #: The resolver will look for any references in all of the imported modules.
+  #: This option requires you to give it the exact name of the reference
+  #: For example, if you `import typing` to use `typing.List`, then referencing #typing.List is valid, but #List is not.
+  #: This also works for `from typing import *`
+  #: Default to `True`.
+  crossref_import: bool = True
+
+  #: Find the reference at all of the project files.
+  #: The resolver will take a look in all modules of the project and fine the reference.
+  #: It will return the closest, top-matched reference.
+  #: The `crossref_import` option requires you to give it the exact module and name of the reference.
+  #: But this is not, you only need to give it the name of the reference.
+  #: Default to `False`.
+  crossref_global: bool = False
+  
```

Comment:
These settings could be an option on the `SmartReferenceResolver` instead. If you make it a `dataclass`, then it can be configured directly through the config, eg.

```suggestion
  resolver: SmartReferenceResolver
```

And in the config

```yml
renderer:
  type: mkdocs
  # ...
  resolver:
    crossref_local: false
```

You could create a copy of it and set the modules in `get_resolver()` or there might also be a cleaner way (e.g. making a `SmartReferenceResolverConfig` the dataclass and give it a `get_resolver()` method that creates a `SmartReferenceResolver` from the config + the modules list).